src/main/kotlin/eu/europa/ec/eudi/pidissuer/adapter/out/diploma/GenerateUserDiplomaData.kt

```diff
@@ -0,0 +1,120 @@
+import kotlinx.serialization.Serializable
+import java.util.Locale.ENGLISH
+
+data class Diploma(
+    val name: String,
+    val achieved: AchievedClaim,
+    val entitledTo: EntitledToClaim? = null,
+    val performedClaim: PerformedClaim? = null,
+)
+
+@Serializable
+data class AchievedClaim(
+    val learningAchievements: List<LearningAchievement>,
+) {
+    companion object : IsAttribute {
+        const val NAME = "achieved"
+        override val attribute: AttributeDetails
+            get() = AttributeDetails(
+                name = NAME,
+                mandatory = false,
+                display = mapOf(
+                    ENGLISH to "An achievement of the person.",
+                ),
+            )
+    }
+}
+
+@Serializable
+data class EntitledToClaim(
+    val entitlements: List<Entitlement>,
+) {
+    companion object : IsAttribute {
+        const val NAME = "entitledTo"
+        override val attribute: AttributeDetails
+            get() = AttributeDetails(
+                name = NAME,
+                mandatory = false,
+                display = mapOf(
+                    ENGLISH to "The entitlement of the person.",
+                ),
+            )
+    }
+}
+
+@Serializable
+data class PerformedClaim(
+    val learningActivities: List<LearningActivity>,
+) {
+    companion object : IsAttribute {
+        const val NAME = "performed"
+        override val attribute: AttributeDetails
+            get() = AttributeDetails(
+                name = NAME,
+                mandatory = false,
+                display = mapOf(
+                    ENGLISH to "A learning activity that a person participated in or attended.",
+                ),
+            )
+    }
+}
+
+/**
+ * The acquisition of knowledge, skills or responsibility and autonomy. A recognised and/or awarded set of learning outcomes of an individual.
+ */
+@Serializable
+data class LearningAchievement(
+    @SerialName("id") val id: String,
+    @SerialName("title") val title: String,
+    @SerialName("definition") val definition: String,
+)
+
+/**
+ * Defines a right, e.g. to practice a profession, take advantage of a learning opportunity or join an organisation,
+ * as a result of the acquisition of knowledge, skills, responsibility and/or autonomy
+ */
+@Serializable
+data class Entitlement(
+    @SerialName("id") val id: String,
+    @SerialName("title") val title: String,
+    @SerialName("definition") val definition: String,
+    @SerialName("issuedDate") val issuedAt: Instant? = null,
+    @SerialName("expiryDate") val expiresOn: Instant? = null,
+)
+
+/**
+ * Defines any process which leads to the acquisition of knowledge, skills or responsibility and autonomy.
+ */
+@Serializable
+data class LearningActivity(
+    @SerialName("id") val id: String,
+    @SerialName("definition") val definition: String,
+    @SerialName("startedAtTime") val started: Instant,
+    @SerialName("endedAtTime") val ended: Instant,
+)
+
+interface GenerateUserDiplomaData {
```

Comment:
Why call it `GenerateUserDiplomaData` and not `GetUserDiplomaData` adding the authorization context as an input parameter (take a look at PID). ?

The fact that we provide an implementation that just generates this is irrelevant, I think